### PR TITLE
Exercise pop up (wip)

### PR DESCRIPTION
**Pull request by AntonChubarov**
Pull request created for integration with exercises cards, markup, styles and some functions will be updated

### CODE REVIEW

File: src/js/exercisePopUp.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
+import { serviceGetExercisesById } from './services.js';
+
+const exercisePopUpBackdrop = document.querySelector('.exercise-pop-up-backdrop');
+const exercisePopUpContent = document.querySelector('.exercise-pop-up-content');
+const exercisePopUpCloseBtn = document.querySelector('.exercise-pop-up-close-btn');
```
Comment (olexandrd) [inline src/js/exercisePopUp.js]: please move to constants.js, refs object

File: src/js/exercisePopUp.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    markup += `
+        <li class="exercise-data-list-item">
+          <p class="exercise-data-parameter-name">Burned Calories</p>
+          <p class="exercise-data-parameter-value">${exercise.burnedCalories}/${exercise.time} min</p>
+        </li>
+      `;
+  }
+
+  return markup;
+}
```
Comment (olexandrd) [inline src/js/exercisePopUp.js]: please move all markup functions to separate js file

**Review by olexandrd — CHANGES_REQUESTED** [review]

File: src/js/exercisePopUp.js (hunk 1 above)
Comment (AntonChubarov) [reply src/js/exercisePopUp.js]: Done

File: src/js/exercisePopUp.js (hunk 2 above)
Comment (AntonChubarov) [reply src/js/exercisePopUp.js]: Done

File: src/js/constants.js
Comment (olexandrd) [inline src/js/constants.js:1]: const refs = {
  list: document.querySelector('.js-filtered-list'),
  filter: document.querySelector('.js-filters'),
  pagination: document.querySelector('.js-pagination'),
  exerciseHeader: document.querySelector('.js-exercise-header'),
  exercisePopUpBackdrop: document.querySelector('.exercise-pop-up-backdrop'),
  exercisePopUpContent: document.querySelector('.exercise-pop-up-content') ...
  
};

File: src/css/exercise_pop_up.css [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+  margin-right: 2px;
+}
+
+.exercise-rating-stars-wrapper {
+  display: inline-flex;
+  gap: 2px;
+  line-height: 1;
+}
+
+.exercise-rating-star {
```
Comment (kazamov) [inline src/css/exercise_pop_up.css]: Remove empty class selector

File: src/js/exercisePopUp.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        refs.exercisePopUpContent.innerHTML = createExerciseMarkup(data);
+        resolve();
+      })
+      .catch(error => {
+        reject(error);
+      });
+  });
+}
+
+// openExercisePopUp('64f389465ae26083f39b17a3');
```
Comment (kazamov) [inline src/js/exercisePopUp.js]: Remove that comment if it is not needed anymore

File: src/js/exercisePopUpMarkup.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+      `;
+  }
+
+  return markup;
+}
+
+const ratingStarSize = 18;
+
+function createRatingStarsMarkup(rating) {
+  const fullStar = `<svg class="exercise-rating-star" width="${ratingStarSize}" height="${ratingStarSize}" viewBox="0 0 32 32"><use href="../images/icons.svg#icon-star" style="fill: var(--yellow);"></use></svg>`;
```
Comment (kazamov) [inline src/js/exercisePopUpMarkup.js:34]: Does existing rating component not fit your needs?

File: src/css/exercise_pop_up.css (hunk 3 above)
Comment (AntonChubarov) [reply src/css/exercise_pop_up.css]: Done

File: src/js/exercisePopUp.js (hunk 4 above)
Comment (AntonChubarov) [reply src/js/exercisePopUp.js]: Done

File: src/js/exercisePopUpMarkup.js (hunk 5 above)
Comment (AntonChubarov) [reply src/js/exercisePopUpMarkup.js:34]: Will reuse component tomorrow

**Review by kazamov — DISMISSED** [review]

**Review by kazamov — APPROVED** [review]

**Review by olexandrd — APPROVED** [review]

File: src/js/constants.js
Comment (AntonChubarov) [reply src/js/constants.js:1]: Done

**Pull request merged**